### PR TITLE
Description Snippet created

### PR DESCRIPTION
**Pull request by kain-ap**
IDescription as general snippet has been created for the note for all the structural Elements of Concepts.

### CODE REVIEW

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+import de.dlr.sc.virsat.build.marker.ui.EsfMarkerImageProvider;
+import de.dlr.sc.virsat.model.dvlm.general.GeneralPackage;
+import de.dlr.sc.virsat.project.markers.IMarkerHelper;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.AUiEStructuralFeatureSectionSnippet;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet;
+
+/**
+ * class ui snippet IDescription implements the interface ui snippet for the description part
+ * @author leps_je
```
Comment (franzTobiasDLR) [inline de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java:38]: Please remove the wrong author tag... We decided to not have any author tags in Virtual Satellite code as thy only become outdated and you can always look up the authors in git

**Review by franzTobiasDLR — APPROVED** [review]
Really cool! Didn't know that we actually have this property in our DVLM model...

**Comment by franzTobiasDLR** [comment]
For others... Here is a screenshot of the new snippet:
grafik

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java (hunk 1 above)
Comment (dellerDLR) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java]: I always wondered where this property is used in the UI :D

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	/**
+	 * Method to set up the Text with specific Grid
+	 * @param editingDomain the editingDomain
+	 */
+	private void setUpText(EditingDomain editingDomain) {
+		GridData gridData = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gridData.heightHint = 100;
+	    textDescription.setLayoutData(gridData);
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java:89]: Check indentation

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			dbCtx.bindValue(WidgetProperties.text(SWT.Modify).observe(textDescription), descriptionProperty.observe(model));
+		}
+	}
+
+	@Override
+	public void updateState(boolean state) {
+		super.updateState(state);
+		
+		Image problemImage = emip.getProblemImageForStructuralFeatureInEobject(model, GeneralPackage.Literals.IDESCRIPTION__DESCRIPTION);
+		if (problemImage == null) {
```
Comment (PhilMFischer) [inline de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java:124]: Is this really code which is individual to this snippet or should it be generic code in one of the super classes?

**Review by PhilMFischer — CHANGES_REQUESTED** [review]
One comment regarding refactoring and one regarding inconsistent indentation of the code. please fix.

**Review by dellerDLR — CHANGES_REQUESTED** [review]
Current version has a conflicting .mailmap, but despite that looks good :)

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java (hunk 2 above)
Comment (kain-ap) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java]: Adjusted the indentation.

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java (hunk 3 above)
Comment (kain-ap) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java]: Not required code in description snippet so removed it

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java (hunk 1 above)
Comment (kain-ap) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetIDescription.java]: Removed the wrong author tag

**Comment by kain-ap** [comment]
Closes #1168

**Pull request merged**